### PR TITLE
Support for Python 3.8

### PR DESCRIPTION
**Pull request by dmgav**
This PR is expected to make the branch compatible with Python 3.8. The changes include:
- removal of use of deprecated feature (exactly the same change was applied in v1.6.0);
- replaced all occurrances of `@asyncio.coroutine` by `async def` (optional change, does not influence the build)

### CODE REVIEW

**Comment by danielballan** [comment]
Nice. Thank you.

File: bluesky/run_engine.py [hunk 1]
```diff
@@ -2399,8 +2367,7 @@ def subscribe(self, func, name='all'):
             self._token_mapping[public_token] = private_tokens
             return public_token
 
-        if name not in DocumentNames:
-            name = DocumentNames[name]
+        name = DocumentNames[name]
```
Comment (mrakitin) [inline bluesky/run_engine.py]: This condition didn't make sense before. Maybe we need to use the following?
```py
if name in DocumentNames:
    name = DocumentNames[name]
```

**Comment by mrakitin** [comment]
We should also fix the `.travis.yml` file as we did for 1.6.x. Here are the warnings from TravisCI:
image

**Comment by dmgav** [comment]
Still fails one test, which passes locally.

File: bluesky/run_engine.py (hunk 1 above)
Comment (dmgav) [reply bluesky/run_engine.py]: This feature was deprecated before and it is removed from Python 3.8. The condition `name in DocumentNames` raises exception with the following message:
```
unsupported operand type(s) for 'in': 'str' and 'EnumMeta'
```
The condition was removed from v1.6.0 completely. I guess the purpose of this condition is to check if `name` is already of `DocumentName` enum type and perform the conversion if not. But the expected values of the 'name' parameter of the function are from the fixed list of strings, so we can perform conversion without verifying if it is a string. The exception will be raised if there is a problem.

**Comment by dmgav** [comment]
The issue was compatibility with PyTest 5.2.1. The test used to pass successfully on the later PyTest 5.3.5. But it seems that using `match` parameter of `pytest.raises` to check the error message is a better way, since it accepts regular expressions.

**Pull request merged**